### ISSUE TITLE
False positive ExplicitJobPermissions on `contents: read` only

### ISSUE THREAD

**Issue by TWiStErRob** [TWiStErRob/net.twisterrob.ghlint#372]
### Version

ghlint 0.5.0

### Reproduction steps

1. File `.github/workflows/x.yml` with the following content:
   ```yaml
   on: push
   permissions:
     contents: read
   jobs:
     repro1:
       runs-on: ubuntu-latest
       steps:
         - run: echo "Problem!"
     repro2:
       runs-on: ubuntu-latest
       steps:
         - run: echo "Problem!"
   ```

2. Run `ghlint` with the following arguments:
   ```shell
   ghlint --console .github/workflows/x.yml
   ```

### Expected behavior

Accept global permissions, it's safe until anything else is added.

### Actual behavior

> repro1 should have explicit permissions
> repro2 should have explicit permissions

Forces permissions on jobs, which could result in a lot of duplication (e.g. 8 jobs all which just checkout and build different things)

### Additional Context

Consider `permissions: {}` to be exempted too.
